code/datums/components/profound_fisher.dm

```diff
@@ -29,32 +26,32 @@
 	if(!HAS_TRAIT(target, TRAIT_FISHING_SPOT))
 		return NONE
 	var/mob/living/living_parent = parent
-	if(!living_parent.CanReach(target))
+	if(living_parent.combat_mode || !living_parent.CanReach(target))
 		return NONE
 	if(living_parent.client)
 		INVOKE_ASYNC(our_rod, TYPE_PROC_REF(/obj/item, melee_attack_chain), parent, target)
 	else
 		INVOKE_ASYNC(src, PROC_REF(pretend_fish), target)
 	return COMPONENT_HOSTILE_NO_ATTACK
 
-/datum/component/profound_fisher/proc/pretend_fish(atom/target)
-	var/fishing_type
-	for(var/type in npc_fishing_preset)
-		if(!istype(target, type))
-			continue
-		fishing_type = npc_fishing_preset[type]
-		break
-	var/datum/fish_source/fish_spot = GLOB.preset_fish_sources[fishing_type]
+/datum/component/profound_fisher/proc/pretend_fish(atom/target, datum/fish_source/fish_spot)
+	var/mob/living/living_parent = parent
+	if(DOING_INTERACTION_WITH_TARGET(living_parent, target))
+		return
+	SEND_SIGNAL(target, COMSIG_NPC_FISHING, args)
```

Comment:
To complete a hat trick I am also side eyeing this really hard as being a getter proc pretending to be an event but can't think of any other better way around what you are trying to do.

Reply:
Why are you passing every arg in if `fish_spot` always reaches this proc as null? Why not just use a bespoke argument list? It seems a little misleading.